tests/Doctrine/Tests/DBAL/Schema/ForeignKeyConstraintTest.php

```diff
@@ -18,12 +32,7 @@ public function testIntersectsIndexColumns(array $indexColumns, $expectedResult)
     {
         $foreignKey = new ForeignKeyConstraint(['foo', 'bar'], 'foreign_table', ['fk_foo', 'fk_bar']);
 
-        $index = $this->getMockBuilder(Index::class)
-            ->disableOriginalConstructor()
-            ->getMock();
-        $index->expects($this->once())
-            ->method('getColumns')
-            ->will($this->returnValue($indexColumns));
+        $index = new Index('INDEX_NAME', $indexColumns);
```

Comment:
I'm vary of test changes here: what was broken here?

Reply:
Nothing, just thought maybe better to use an actual object instead of mock no?

Reply:
If the previous code works without changes, let's avoid changing it at all. Let's put test refactoring in separate patches, to avoid hiding potential BC breaks

Reply:
@Ocramius as you can see tests fail and now I remember why. Because I use `getUnquotedColumns` instead of `getColumns` and I added some columns with quotes into data provider. And if I mock here can't test quoted names. I mean this:
```
$index->expects($this->once())
            ->method('getUnquotedColumns')
            ->will($this->returnValue($indexColumns));
``` 
should always return unquoted names and I can't add quoted names in data provider but maybe I don't have to?
 I still think actual implementation would be better here and anyway I have to change mocked method anyway.

Reply:
@Ocramius if you want I can still leave this mock but will have to change mocked method and will also have to duplicate this test and test data to actually test quoted names.